src/librustc_trans/glue.rs

```diff
@@ -205,9 +204,14 @@ pub fn implement_drop_glue<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>, g: DropGlueKi
     // non-null, (and maybe we need to continue doing so), but we now
     // must definitely check for special bit-patterns corresponding to
     // the special dtor markings.
-    let v0 = get_param(llfn, 0);
     let t = g.ty();
 
+    let args = if ccx.shared().type_is_sized(t) {
+        MaybeSizedValue::sized(get_param(llfn, 0))
```

Comment:
I'd move the first argument out (`data`), and also call this `v` or `value` or `ptr`, not `args`.